### PR TITLE
SONARSCALA-5 Reuse the same build artifact through CI pipeline

### PR DESCRIPTION
**Pull request by dorian-burihabwa-sonarsource**

### CODE REVIEW

File: .cirrus.yml [hunk 1]
```diff
@@ -16,11 +23,8 @@ build_task:
     cpu: 8
     memory: 4G
   env:
-    # Possible values for ARTIFACTORY_DEPLOY_REPO: sonarsource-private-qa, sonarsource-public-qa
-    ARTIFACTORY_DEPLOY_REPO: FIXME
-    ARTIFACTORY_DEPLOY_USERNAME: VAULT[development/artifactory/token/${CIRRUS_REPO_OWNER}-${CIRRUS_REPO_NAME}-qa-deployer username]
-    ARTIFACTORY_DEPLOY_PASSWORD: VAULT[development/artifactory/token/${CIRRUS_REPO_OWNER}-${CIRRUS_REPO_NAME}-qa-deployer access_token]
-    DEPLOY_PULL_REQUEST: "true"
+    SONAR_TOKEN: FIXME
+    SONAR_HOST_URL: FIXME
```
Comment (ADarko22) [inline .cirrus.yml:27]: It should be possible to use Next now. First create the Project from SPEED
```suggestion
    SONAR_TOKEN: VAULT[development/kv/data/next data.token]
    SONAR_HOST_URL: VAULT[development/kv/data/next data.url]
```

File: .cirrus.yml [hunk 2]
```diff
@@ -29,7 +33,15 @@ build_task:
     populate_script: mkdir -p ${GRADLE_USER_HOME}/caches
   build_script:
     - source cirrus-env BUILD
-    - ./gradlew build
+    - function gradle(){ ./gradlew "$@"; }; export -f gradle
+    - gradle --version
+    - source set_gradle_build_version
+    - regular_gradle_build_deploy_analyze --exclude-task sonar
```
Comment (ADarko22) [inline .cirrus.yml:39]: Once the analysis on Next is configure we can trigger it:
```suggestion
    - regular_gradle_build_deploy_analyze 
```

File: .cirrus.yml [hunk 3]
```diff
@@ -53,8 +60,6 @@ qa_plugin_task:
       - echo $CIRRUS_OS
       - cat **/*.gradle* **/gradle*.properties || true
     populate_script: mkdir -p ${GRADLE_USER_HOME}/caches
```
Comment (ADarko22) [inline .cirrus.yml:62]: In sonar-go we're following the sonar-kotlin approach to create a template for the gradle cache; see [this PR](https://github.com/SonarSource/sonar-go/pull/2/files )

File: .cirrus.yml (hunk 2 above)
Comment (dorian-burihabwa-sonarsource) [reply .cirrus.yml]: I will fix that in another PR when tackling SONARSCALA-4

File: .cirrus.yml (hunk 1 above)
Comment (dorian-burihabwa-sonarsource) [reply .cirrus.yml:38]: I will solve this in another PR as I have to create the next project first

**Review by ADarko22 — APPROVED** [review]

**Pull request merged**